tools/stats_utils/s3_stat_parser.py

```diff
@@ -127,3 +133,28 @@ def get_cases(
         else:
             raise RuntimeError(f'Unknown format version: {version}')
     return cases
+
+
+def _parse_s3_summaries(summaries: Any, jobs: List[str]) -> Dict[str, List[Any]]:
+    summary_dict = defaultdict(list)
+    for summary in summaries:
+        summary_job = summary.key.split('/')[2]
+        if summary_job in jobs or len(jobs) == 0:
+            binary = summary.get()["Body"].read()
+            string = bz2.decompress(binary).decode("utf-8")
+            summary_dict[summary_job].append(json.loads(string))
+    return summary_dict
+
+# Collect and decompress S3 test stats summaries into JSON.
+# data stored on S3 buckets are pathed by {sha}/{job} so we also allow
+# optional jobs filter
+def get_test_stats_summaries(*, sha: str, jobs: Optional[List[str]] = None) -> Dict[str, List[Any]]:
+    bucket = get_S3_bucket_readonly(OSSCI_METRICS_BUCKET)
+    summaries = list(bucket.objects.filter(Prefix=f"test_time/{sha}"))
+    return _parse_s3_summaries(summaries, jobs=list(jobs or []))
+
+
+def get_test_stats_summaries_for_job(*, sha: str, job_prefix: str) -> Dict[str, List[Any]]:
+    bucket = get_S3_bucket_readonly(OSSCI_METRICS_BUCKET)
+    summaries = list(bucket.objects.filter(Prefix=f"test_time/{sha}/{job_prefix}"))
```

Comment:
This is potentially very large, if job_prefix is not specified.

Reply:
It might be fine just to not have the list()?
```suggestion
    summaries = bucket.objects.filter(Prefix=f"test_time/{sha}/{job_prefix}")
```

Reply:
Actually not really too large since sha is required as a string not a list

Reply:
Nice, but job_prefix could still be empty string :p 

The current implementation is good.